### DISCUSSION TITLE
Getting started with TailwindCSS

### DISCUSSION THREAD

**Discussion by paulguilloux** [keycloakify/keycloakify#578] [Q&A] [answered]
Hi, I'm having a hard time getting TailwindCSS to work inside Keycloakify.

What I tried : 
- Installed `tailwindcss` with Yarn
- Set up tailwind.config.js with `content: ["./src/**/*.{html,js,jsx,ts,tsx}"],`
- Created `/src/main.css`
- Imported this css file in `src/main.tsx`
- Ejected Template.tsx
- Remove other css imports in Template.tsx (where `areAllStyleSheetsLoaded` is defined)

I am getting no luck with this : the old CSS is correctly removed, but the Tailwind default style doesn't apply. The font remains Times New Roman, etc.

Are there any examples/resources that I can look into to get started with Tailwindcss in Keycloakify ?

Thanks
Paul

**Comment by garronej** [comment, marked as answer]
Hello @paulguilloux,  

I haven't had time to redact the tutorial yet.  
However tailwind works in Keycloakify project like it does in any other Vite app.  

See Tailwind getting started with Vite: https://tailwindcss.com/docs/guides/vite  

I've created a branch on the starter template to help you get started:  

<img width="1066" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/3f06a287-03e3-4441-a4e9-e4ea6b76f388">

Here is the branch: https://github.com/keycloakify/keycloakify-starter/tree/tailwind  

Here is the commit where I added Tailwind: https://github.com/keycloakify/keycloakify-starter/commit/8d44f2c0f4adbe197420b7fb1b2ae858b47f9372  

The only thing that might not be obvious is, if you want to customize the style without having to eject the pages you have to use [Tailwind Merge](https://github.com/dcastil/tailwind-merge) like shown here:  

https://github.com/keycloakify/keycloakify-starter/blob/8d44f2c0f4adbe197420b7fb1b2ae858b47f9372/src/login/KcPage.tsx#L54-L58  

I hope it helps

**Reply by paulguilloux** [reply]
This is great, thank you so much for the help! Keycloakify is a great product and this makes it even more powerful :)

**Reply by garronej** [reply]
Hey @paulguilloux, I've updated the tailwind documentation you might want to check it out: https://docs.keycloakify.dev/customization-strategies/css-level-customization/using-tailwind

**Comment by garronej** [comment]
If you want to disable the default style ([patternlyfly](https://www.patternfly.org/)) on the pages you have ejected you can simply set `doUseDefaultCss` to `false`:  

<img width="590" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/242738f1-abc6-4910-a28d-460a1587c66d">

But it's a lot of work to re-style everything:  

<img width="1148" alt="image" src="https://github.com/keycloakify/keycloakify/assets/6702424/7c44688b-1af9-4473-8434-26e44d841c24">

You might want to favor overriding the default styles.
